program/rust/src/tests/test_upd_price_v2.rs

```diff
@@ -429,16 +421,86 @@ fn test_upd_price_v2() -> Result<(), Box<dyn std::error::Error>> {
+        assert_eq!(price_data.price_cumulative.num_down_slots, 16);
+    }
+
+    // add new test for multiple publishers and ensure that price_cumulative is updated correctly
+    let mut funding_setup_two = AccountSetup::new_funding();
+    let funding_account_two = funding_setup_two.as_account_info();
+
+    {
+        let mut price_data = load_checked::<PriceAccount>(&price_account, PC_VERSION).unwrap();
+        price_data.num_ = 2;
+        price_data.comp_[1].pub_ = *funding_account_two.key;
+    }
+
+    populate_instruction(&mut instruction_data, 10, 1, 100);
+    update_clock_slot(&mut clock_account, 100);
+    process_instruction(
+        &program_id,
+        &[
+            funding_account.clone(),
+            price_account.clone(),
+            clock_account.clone(),
+        ],
+        &instruction_data,
+    )?;
 
+    populate_instruction(&mut instruction_data, 20, 2, 100);
+    process_instruction(
+        &program_id,
+        &[
+            funding_account_two.clone(),
+            price_account.clone(),
+            clock_account.clone(),
+        ],
+        &instruction_data,
+    )?;
+
+    {
+        let price_data = load_checked::<PriceAccount>(&price_account, PC_VERSION).unwrap();
+        assert_eq!(price_data.comp_[0].latest_.price_, 10);
+        assert_eq!(price_data.comp_[0].latest_.conf_, 1);
+        assert_eq!(price_data.comp_[0].latest_.pub_slot_, 100);
+        assert_eq!(price_data.comp_[0].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.comp_[1].latest_.price_, 20);
+        assert_eq!(price_data.comp_[1].latest_.conf_, 2);
+        assert_eq!(price_data.comp_[1].latest_.pub_slot_, 100);
+        assert_eq!(price_data.comp_[1].latest_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.valid_slot_, 100);
+        assert_eq!(price_data.agg_.pub_slot_, 100);
+        assert_eq!(price_data.agg_.price_, 14);
+        assert_eq!(price_data.agg_.conf_, 6);
+        assert_eq!(price_data.agg_.status_, PC_STATUS_TRADING);
+        assert_eq!(price_data.prev_slot_, 51);
+        assert_eq!(price_data.prev_price_, 55);
+        assert_eq!(price_data.prev_conf_, 5);
+        assert_eq!(
+            price_data.prev_price_cumulative.price,
+            42 + 81 * 4 - 100 * 4 + 60 * 41 + 55
+        );
+        assert_eq!(
+            price_data.prev_price_cumulative.conf,
+            2 * 5 + 4 + 41 * 4 + 5
+        );
+        assert_eq!(price_data.prev_price_cumulative.num_down_slots, 16);
         assert_eq!(
             price_data.price_cumulative.price,
-            3 * 42 + 81 * 3 - 100 * 3 + 42 * 60
+            42 + 81 * 4 - 100 * 4 + 60 * 41 + 55 + 14 * 49
+        ); // (42 + 81 * 4 - 100 * 4 + 60 * 41 + 55) is the price cumulative from the previous test and 14 * 49 (slot_gap) is the price cumulative from this test
+        assert_eq!(
+            price_data.price_cumulative.conf,
+            2 * 5 + 4 + 41 * 4 + 5 + 6 * 49
```

Comment:
Feels like you should test the behavior when there are two aggregations in one slot here

Reply:
what do you mean specifically? this test is meant to simulate 2 aggregations in one slot

Reply:
Yes but no asserts after the first aggregation

Reply:
will add asserts